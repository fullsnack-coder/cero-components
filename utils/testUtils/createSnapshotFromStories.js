import React from 'react'
import { render } from '@testing-library/react'
import { composeStories } from '@storybook/testing-react'

// TODO: Move this to a generic approach where it
// reads all the stories and works as Storyshoot
const createSnapshotFromStories = (componentName, stories) => {
  const composedStories = composeStories(stories)

  Object.entries(composedStories).forEach(([story, Component]) => {
    describe(`when '${componentName}' ${story} is mounted`, () => {
      it('should render', () => {
        const { asFragment } = render(<Component />)
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
}

export default createSnapshotFromStories
