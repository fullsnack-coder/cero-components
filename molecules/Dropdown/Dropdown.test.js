import * as stories from './Dropdown.stories'
import createSnapshotFromStories from '../../utils/testUtils/createSnapshotFromStories'

describe('[ molecules / Dropdown ]', () => {
  describe('#Dropdown', () => createSnapshotFromStories('Dropdown', stories))
})
