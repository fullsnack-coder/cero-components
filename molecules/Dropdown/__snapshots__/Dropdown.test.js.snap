// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ molecules / Dropdown ] #Dropdown when 'Dropdown' Default is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="dropdown"
  >
    <div
      class="dropdown-icon icon color-base size-md background-highlight"
      style="width: 35px; height: 35px;"
    >
      <svg
        height="35"
        viewBox="0 0 15 9"
        width="35"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M7.82633 8.70623L0.882568 1.54063L1.89405 0.496826L7.82633 6.61863L13.7586 0.496826L14.7701 1.54063L7.82633 8.70623Z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <select
      class="dropdown-select"
    >
      <option
        value="800"
      >
        8:00 AM
      </option>
      <option
        value="1300"
      >
        1:00 PM
      </option>
    </select>
  </div>
</DocumentFragment>
`;

exports[`[ molecules / Dropdown ] #Dropdown when 'Dropdown' Inline is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="dropdown is-inline"
  >
    <div
      class="dropdown-icon icon color-base size-md background-highlight"
      style="width: 35px; height: 35px;"
    >
      <svg
        height="35"
        viewBox="0 0 15 9"
        width="35"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M7.82633 8.70623L0.882568 1.54063L1.89405 0.496826L7.82633 6.61863L13.7586 0.496826L14.7701 1.54063L7.82633 8.70623Z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <select
      class="dropdown-select"
    >
      <option
        value="800"
      >
        8:00 AM
      </option>
      <option
        value="1300"
      >
        1:00 PM
      </option>
    </select>
  </div>
</DocumentFragment>
`;

exports[`[ molecules / Dropdown ] #Dropdown when 'Dropdown' Value is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="dropdown"
  >
    <div
      class="dropdown-icon icon color-base size-md background-highlight"
      style="width: 35px; height: 35px;"
    >
      <svg
        height="35"
        viewBox="0 0 15 9"
        width="35"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M7.82633 8.70623L0.882568 1.54063L1.89405 0.496826L7.82633 6.61863L13.7586 0.496826L14.7701 1.54063L7.82633 8.70623Z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <select
      class="dropdown-select"
    >
      <option
        value="800"
      >
        8:00 AM
      </option>
      <option
        value="1300"
      >
        1:00 PM
      </option>
    </select>
  </div>
</DocumentFragment>
`;
